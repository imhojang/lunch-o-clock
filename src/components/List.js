import React from 'react';

const List = props => {
  const renderedList = props.list.map(item => {
    return <div key={item._id}>{item.name}</div>;
  });

  return <div>{renderedList}</div>;
};

export default List;
