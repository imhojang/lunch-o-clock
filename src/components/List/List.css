ul {
  padding: 0;
  margin-bottom: 0;
}

li {
  height: 36px;
  list-style: none;
  background-color: rgba(0, 0, 0, 0.05);
  border-radius: 3px;
  border: 1px solid #ccc;
  margin: 5px 0;
  padding: 5px;
  position: relative;
}

li > span {
  display: inline-block;
  width: 200px;
  text-overflow: ellipsis;
}

li > svg {
  position: absolute;
  top: 9px;
  right: 9px;
  cursor: pointer;
  width: 16px;
  height: 16px;
}

li > svg:hover {
  fill: white;
}

li > button:active {
  fill: #ffd32e;
}
