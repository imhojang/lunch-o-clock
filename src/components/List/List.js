import React from 'react';
import './List.css';
import { DeleteIcon } from '../../utils/Icons';

const List = ({ items, handleDelete }) => {
  const renderedList = items.map(item => {
    return (
      <li key={item._id}>
        <span>{item.name}</span>
        {handleDelete && (
          <DeleteIcon
            className='remove-person-button'
            handleClick={handleDelete}
            item={item}
          />
        )}
      </li>
    );
  });

  return <ul>{renderedList}</ul>;
};

export default List;
