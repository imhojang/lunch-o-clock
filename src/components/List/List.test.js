import React from 'react'
import { shallow, mount } from 'enzyme'
import List from './List'

describe('List structure', () => {
  let list
  let items

  it('should throw without items prop', () => {
    items = null
    expect(() => shallow(<List items={items} />)).toThrow()
  })

  it('renders correctly with items prop', () => {
    items = [
      { name: 'item 1', _id: 'abc' },
      { name: 'item 2', _id: 'def' }
    ]
    list = shallow(<List items={items} />)
  })

  it('matches snapshot', () => {
    expect(list).toMatchSnapshot()
  })

  it('renders an unordered list correctly', () => {
    expect(list.find('ul')).toHaveLength(1)
  })

  it('renders correct number of span element ', () => {
    expect(list.find('span')).toHaveLength(2)
  })

  it('renders correct number of list item', () => {
    expect(list.find('li')).toHaveLength(2)
  })
})

describe('List behavior', () => {
  let list
  let items
  let handleDelete

  beforeEach(() => {
    items = [{ name: 'item 1', _id: 'abc' }]
    handleDelete = jest.fn()
    list = mount(<List items={items} handleDelete={handleDelete} />)
  })

  it('props are passed down correctly', () => {
    expect(list.props().handleDelete).toBe(handleDelete)
    expect(list.props().items).toBe(items)
  })

  it('list item renders a svg that acts as a button', () => {
    const button = list.find('svg').first()
    expect(button).toBeDefined()
  })

  it('delete button clicks calls handleDelete correctly', () => {
    const button = list.find('svg').first()
    button.simulate('click')
    expect(handleDelete).toHaveBeenCalled()
  })
})
