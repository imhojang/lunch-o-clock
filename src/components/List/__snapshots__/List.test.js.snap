// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List structure matches snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <List
    items={
        Array [
            Object {
              "_id": "abc",
              "name": "item 1",
            },
            Object {
              "_id": "def",
              "name": "item 2",
            },
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <li>
          <span>
                    item 1
          </span>
</li>,
        <li>
          <span>
                    item 2
          </span>
</li>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "abc",
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <span>
              item 1
</span>,
            undefined,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "item 1",
            },
            "ref": null,
            "rendered": "item 1",
            "type": "span",
          },
          undefined,
        ],
        "type": "li",
      },
      Object {
        "instance": null,
        "key": "def",
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <span>
              item 2
</span>,
            undefined,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "item 2",
            },
            "ref": null,
            "rendered": "item 2",
            "type": "span",
          },
          undefined,
        ],
        "type": "li",
      },
    ],
    "type": "ul",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <li>
            <span>
                        item 1
            </span>
</li>,
          <li>
            <span>
                        item 2
            </span>
</li>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "abc",
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <span>
                item 1
</span>,
              undefined,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "item 1",
              },
              "ref": null,
              "rendered": "item 1",
              "type": "span",
            },
            undefined,
          ],
          "type": "li",
        },
        Object {
          "instance": null,
          "key": "def",
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <span>
                item 2
</span>,
              undefined,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "item 2",
              },
              "ref": null,
              "rendered": "item 2",
              "type": "span",
            },
            undefined,
          ],
          "type": "li",
        },
      ],
      "type": "ul",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
}
`;
